gui/src/app/pages/HomePage/DataGenerationWindow/DataPyFileEditor.tsx

```diff
@@ -0,0 +1,103 @@
+import TextEditor, { ToolbarItem } from "@SpComponents/TextEditor";
+import { writeConsoleOutToDiv } from "app/pyodide/AnalysisPyFileEditor";
+import { PyodideWorkerStatus } from "app/pyodide/pyodideWorker/pyodideWorkerTypes";
+import usePyodideWorker from "app/pyodide/pyodideWorker/usePyodideWorker";
+import {
+  FunctionComponent,
+  RefObject,
+  useCallback,
+  useMemo,
+  useState,
+} from "react";
+import getDataGenerationToolbarItems from "./getDataGenerationToolbarItems";
+
+type Props = {
+  fileName: string;
+  fileContent: string;
+  onSaveContent: () => void;
+  editedFileContent: string;
+  setEditedFileContent: (text: string) => void;
+  readOnly: boolean;
+  setData?: (data: any) => void;
+  outputDiv: RefObject<HTMLDivElement>;
+};
+
+const DataPyFileEditor: FunctionComponent<Props> = ({
+  fileName,
+  fileContent,
+  onSaveContent,
+  editedFileContent,
+  setEditedFileContent,
+  setData,
+  readOnly,
+  outputDiv,
+}) => {
+  const [status, setStatus] = useState<PyodideWorkerStatus>("idle");
```

Comment:
Funny that I never thought about this before, but is it weird that the worker status is something tracked by a main-thread UI component rather than the component querying the worker for its status?

I'm sure this actually makes sense (esp. b/c of the worker thread going nonresponsive while it's actually running a job) but it just stuck out to me.

Reply:
I know what you mean, but I think it's important for the status state to be synchronously available in the main thread.